### DISCUSSION TITLE
Network Load Balancer Health check

### DISCUSSION THREAD

**Discussion by bigbits22** [istio/istio#49906] [General]
Hello,
I have deployed istio 1.20 (and 1.18) on Oracle Cloud and I have configured istio-ingressgateway with Network Load Balancer.
It work very well but in Oracle console I see Load Balancer Health in error because only one node behind balancer can respond to health check.
If I try to scale up ingress manually, the health check error disappear.
What's correct way to configure? Should I increase manually the input as well as the number of nodes?
Thank you

**Comment by abasitt** [comment]
more details please. 
what port are you using for healthcheck? and how are these ports exposed in oracle cloud? Is it nodeports ? what is the externaltrafficpolicy type of istio-ingress service?

**Reply by bigbits22** [reply]
Hello abasitt,
thanks to your answer I discovered that the problems it's "External Traffic Policy".
If this variable it's "Cluster", health check work.
If this variable it's "Local", health check not work.
My goal is to be able to keep the source ip address in the logs.
Now, I try to investigate.

**Reply by abasitt** [reply]
Yeah with nodeport and external traffic policy local, its going to be difficult unless oracle cloud nlb controller natively support this.
You can also try other methods to preserver source ip e.g. https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/

**Reply by bigbits22** [reply]
I have see this important note on Istio documentation
![immagine](https://github.com/istio/istio/assets/101560721/56f355e5-3ebf-4d1a-9d0b-9be4c0a82e9a)

How I can automatically scale ingress pod to number of nodes? Today I have only 2 nodes but it can scale up automatically.

**Reply by bigbits22** [reply]
I found this solution (deploy ingress like DaemonSet) https://stackoverflow.com/questions/67373027/istio-run-ingress-gateway-on-every-node
In your opinion is this correct?

**Reply by abasitt** [reply]
yes running ingress pod as daemon-set is totally valid design as well and people do that with similar requirements like yours.

**Reply by stanyzra** [reply]
Running ingress pod as daemon-set indeed resolved the health check for me too using Oracle Cloud. I actually have a question for @bigbits22, have you experienced any issues with the NLB after configuring it like daemon-set? I've tried to use Oracle's NLB with Istio in the past and for some reason the NLB randomly stops working and I couldn't figure out why. Back in that time, the Oracle console always pointed my NLB as unhealthy because I didn't have a ingress pod running in every node of my cluster, but even with that the NLB was still working, the problem is that, as I said, it randomly stops working and the only way to make it back to normal was editing some configuration in the ingress service or completly recreating the NLB. Have you experienced any similar thing now with daemon-set configuration?
